packages/labs/nextjs/src/lib/apply-dsd-polyfill.ts

```diff
@@ -0,0 +1,13 @@
+/// <reference lib="dom" />
+
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {hydrateShadowRoots} from '@webcomponents/template-shadowroot';
+
+if (!HTMLTemplateElement.prototype.hasOwnProperty('shadowRootMode')) {
```

Comment:
Should this be conditional, or always applied?

Reply:
This is the feature detection so the ponyfill is only applied for browsers that don't support it. https://developer.chrome.com/articles/declarative-shadow-dom/#feature-detection-and-browser-support